lib/src/session.rs

```diff
@@ -658,6 +658,9 @@ impl Session {
     }
 }
 
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
+pub struct AutoDecompressResponse {}
```

Comment:
This is a cool way to represent the flag, but there is a drawback for the future: we'll need to add additional metadata to requests and responses to keep up with C@E proper. (I'm doing some work that expands those out right now).

I think we should consider more closely following C@E's approach of introducing `FastlyRequestMetadata`/`FastlyResponseMetadata` structs with fields representing the various knobs.

Reply:
Done in 7aa1e6d ! Except that I called it `ViceroyRequestMetadata`.